contrib/vagrant/provision-util.sh

```diff
@@ -110,23 +166,120 @@ os::util::get-network-plugin() {
+    if [ "${dind_management_script}" = "false" ]; then
+      if [ "${plugin}" != "" ]; then
         >&2 echo "Invalid network plugin: ${plugin}"
+      fi
+      >&2 echo "Using default network plugin: ${default_plugin}"
     fi
-    >&2 echo "Using default network plugin: ${default_plugin}"
     plugin="${default_plugin}"
   fi
   echo "${plugin}"
 }
 
-os::util::wait-for-condition() {
-  local start_msg=$1
-  local error_msg=$2
+os::provision::base-provision() {
+  os::provision::fixup-net-udev
+
+  os::provision::setup-hosts-file "${MASTER_NAME}" "${MASTER_IP}" NODE_NAMES \
+    NODE_IPS
+
+  os::provision::install-pkgs
+}
+
+os::provision::fixup-net-udev() {
+  if [ "${FIXUP_NET_UDEV}" == "true" ]; then
+    NETWORK_CONF_PATH=/etc/sysconfig/network-scripts/
+    rm -f ${NETWORK_CONF_PATH}ifcfg-enp*
+    if [[ -f "${NETWORK_CONF_PATH}ifcfg-eth1" ]]; then
+      sed -i 's/^NM_CONTROLLED=no/#NM_CONTROLLED=no/' ${NETWORK_CONF_PATH}ifcfg-eth1
+      if ! grep -q "NAME=" ${NETWORK_CONF_PATH}ifcfg-eth1; then
+        echo "NAME=openshift" >> ${NETWORK_CONF_PATH}ifcfg-eth1
+      fi
+      nmcli con reload
+      nmcli dev disconnect eth1
+      nmcli con up "openshift"
+    fi
+  fi
+}
+
+os::provision::in-container() {
+  test -f /.dockerinit
+}
+
+os::provision::install-pkgs() {
+  # Only install packages if not deploying to a container.  A
+  # container is expected to have installed packages as part of image
+  # creation.
+  if ! os::provision::in-container; then
+    yum install -y deltarpm
+    yum update -y
+    yum install -y docker-io git golang e2fsprogs hg net-tools bridge-utils which ethtool
+
+    systemctl enable docker
+    systemctl start docker
+  fi
+}
+
+os::provision::start-os-service() {
+  local unit_name=$1
+  local description=$2
+  local exec_start=$3
+  local work_dir=${4:-${CONFIG_ROOT}/}
+
+  local dind_env_var=
+  if os::provision::in-container; then
+    dind_env_var="OPENSHIFT_DIND=true"
+  fi
+
+  cat <<EOF > "/usr/lib/systemd/system/${unit_name}.service"
+[Unit]
+Description=${description}
+Requires=network.target
+After=docker.target network.target
+
+[Service]
+Environment=${dind_env_var}
+ExecStart=${exec_start}
+WorkingDirectory=${work_dir}
+Restart=on-failure
+RestartSec=10s
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+  systemctl daemon-reload > /dev/null
+  systemctl enable "${unit_name}.service" &> /dev/null
+  systemctl start "${unit_name}.service"
+}
+
+os::provision::start-node-service() {
+  local config_root=$1
+  local node_name=$2
+
+  # Copy over the certificates directory so that each node has a copy.
+  cp -r "${config_root}/openshift.local.config" /
+  if [ -d /home/vagrant ]; then
+    chown -R vagrant.vagrant /openshift.local.config
+  fi
+
+  cmd="/usr/bin/openshift start node --loglevel=${LOG_LEVEL} \
+--config=$(os::provision::get-node-config ${config_root} ${node_name})"
+  os::provision::start-os-service "openshift-node" "OpenShift Node" "${cmd}" /
+}
+
+OS_WAIT_FOREVER=-1
```

Comment:
A vm node of a devcluster waits forever for its configuration to be generated.  Given the huge variance in how long it will take for a master to install packages and perform go builds, is there any reason not to wait indefinitely?